### PR TITLE
fix: GithubAuth should return empty Optional when authentication is failed

### LINKED ISSUE #406: Github authentication error should not cause 500 error
When using Github authentication, auth errors from Github API causes 500 error for user. We should handle authentication errors and return proper status code in that case 401.

```
artipie         | Caused by: java.lang.AssertionError: HTTP response status is not equal to 200:
artipie         | 401 Unauthorized [https://api.github.com/user]
artipie         | Server: GitHub.com
artipie         | Date: Thu, 30 Jul 2020 14:39:46 GMT
artipie         | Content-Type: application/json; charset=utf-8
artipie         | Content-Length: 83
artipie         | Status: 401 Unauthorized
artipie         | X-GitHub-Media-Type: github.v3
artipie         | X-RateLimit-Limit: 60
artipie         | X-RateLimit-Remaining: 55
artipie         | X-RateLimit-Reset: 1596123034
artipie         | Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset
artipie         | Access-Control-Allow-Origin: *
artipie         | Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
artipie         | X-Frame-Options: deny
artipie         | X-Content-Type-Options: nosniff
artipie         | X-XSS-Protection: 1; mode=block
artipie         | Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
artipie         | Content-Security-Policy: default-src 'none'
artipie         | Vary: Accept-Encoding, Accept, X-Requested-With
artipie         | 
artipie         | {"message":"Bad credentials","documentation_url":"https://developer.github.com/v3"}
artipie         | Expected: HTTP response with status 200
artipie         |      but: was <401 Unauthorized [https://api.github.com/user]
artipie         | Server: GitHub.com
artipie         | Date: Thu, 30 Jul 2020 14:39:46 GMT
artipie         | Content-Type: application/json; charset=utf-8
artipie         | Content-Length: 83
artipie         | Status: 401 Unauthorized
artipie         | X-GitHub-Media-Type: github.v3
artipie         | X-RateLimit-Limit: 60
artipie         | X-RateLimit-Remaining: 55
artipie         | X-RateLimit-Reset: 1596123034
artipie         | Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset
artipie         | Access-Control-Allow-Origin: *
artipie         | Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
artipie         | X-Frame-Options: deny
artipie         | X-Content-Type-Options: nosniff
artipie         | X-XSS-Protection: 1; mode=block
artipie         | Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
artipie         | Content-Security-Policy: default-src 'none'
artipie         | Vary: Accept-Encoding, Accept, X-Requested-With
artipie         | 
artipie         | {"message":"Bad credentials","documentation_url":"https://developer.github.com/v3"}>
artipie         | 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
artipie         | 	at com.jcabi.http.response.RestResponse.assertStatus(RestResponse.java:111)
artipie         | 	at com.jcabi.github.RtJson.fetch(RtJson.java:77)
artipie         | 	at com.jcabi.github.RtUser.json(RtUser.java:166)
artipie         | 	at com.jcabi.github.RtUser.login(RtUser.java:121)
artipie         | 	at com.artipie.auth.GithubAuth.lambda$new$0(GithubAuth.java:65)
artipie         | 	at com.artipie.auth.GithubAuth.user(GithubAuth.java:86)
artipie         | 	at com.artipie.auth.ChainedAuth.user(ChainedAuth.java:74)
artipie         | 	at com.artipie.auth.LoggingAuth.user(LoggingAuth.java:68)
artipie         | 	at com.artipie.http.auth.BasicIdentities.lambda$user$3(BasicIdentities.java:67)
artipie         | 	at java.base/java.util.Optional.flatMap(Unknown Source)
artipie         | 	at com.artipie.http.auth.BasicIdentities.user(BasicIdentities.java:67)
artipie         | 	at com.artipie.http.auth.SliceAuth.response(SliceAuth.java:85)
artipie         | 	at com.artipie.http.rt.RtRulePath.response(RtRulePath.java:72)
artipie         | 	at com.artipie.http.rt.SliceRoute.lambda$response$0(SliceRoute.java:86)
artipie         | 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
artipie         | 	at java.base/java.util.Spliterators$IteratorSpliterator.tryAdvance(Unknown Source)
artipie         | 	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(Unknown Source)
artipie         | 	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(Unknown Source)
artipie         | 	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
artipie         | 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
artipie         | 	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(Unknown Source)
artipie         | 	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
artipie         | 	at java.base/java.util.stream.ReferencePipeline.findFirst(Unknown Source)
artipie         | 	at com.artipie.http.rt.SliceRoute.response(SliceRoute.java:89)
artipie         | 	at com.artipie.http.Slice$Wrap.response(Slice.java:81)
artipie         | 	at com.artipie.http.slice.TrimPathSlice.response(TrimPathSlice.java:112)
artipie         | 	at com.artipie.http.async.AsyncSlice.lambda$response$0(AsyncSlice.java:63)
artipie         | 	... 15 more
```

### PR DESCRIPTION
**Pull request by dgarus**
Closes #406 

- added auth error handling to `GithubAuth`
- added tests for cases when auth is failed and any other error happens

### CODE REVIEW

**Review by olenagerasimova — DISMISSED** [review]
@dgarus thanks
@g4s8 please, check/merge

File: src/main/java/com/artipie/auth/GithubAuth.java [hunk 1]
```diff
@@ -64,11 +67,23 @@ public GithubAuth() {
...
+                }
+            } catch (final AssertionError error) {
+                if (
+                    new Unchecked<>(
+                        new Not(
+                            new Contains(error.getMessage(), "401 Unauthorized")
+                        )
+                    ).value()
+                ) {
+                    throw error;
```
Comment (g4s8) [inline src/main/java/com/artipie/auth/GithubAuth.java:85]: I think throwing `AssertionError` is not really a good place for production code with disabled assertions (`java -ea` flag is not enabled). Maybe we can re-throw it with some different type?

**Review by g4s8 — CHANGES_REQUESTED** [review]
@dgarus thanks, please check my comments

File: src/main/java/com/artipie/auth/GithubAuth.java [hunk 2]
```diff
@@ -64,11 +67,23 @@ public GithubAuth() {
...
+                ) {
+                    result = Optional.of(new Authentication.User(login));
+                }
+            } catch (final AssertionError error) {
+                if (
+                    new Unchecked<>(
+                        new Not(
+                            new Contains(error.getMessage(), "401 Unauthorized")
+                        )
+                    ).value()
```
[the remark this answers is not in the document]
Comment (dgarus) [reply src/main/java/com/artipie/auth/GithubAuth.java]: Sure, I agree.

File: src/main/java/com/artipie/auth/GithubAuth.java (hunk 1 above)
Comment (dgarus) [reply src/main/java/com/artipie/auth/GithubAuth.java]: Ok

**Comment by dgarus** [comment]
@olenagerasimova thanks for the review.
@g4s8 thanks, I fixed your comments.

**Review by g4s8 — APPROVED** [review]
@dgarus thanks

**Comment by dgarus** [comment]
@g4s8, thank you for the review!
Could you please merge the PR?

**Pull request merged**